Mathlib/Probability/Kernel/Disintegration.lean

```diff
@@ -369,6 +369,32 @@ theorem kernel.const_eq_compProd (γ : Type*) [MeasurableSpace γ] (ρ : Measure
     kernel.ext_iff'.mp (kernel.const_unit_eq_compProd ρ) () s hs
 #align probability_theory.kernel.const_eq_comp_prod ProbabilityTheory.kernel.const_eq_compProd
 
+/-- If the singleton `{x}` has non-zero mass for `ρ.fst`, then for all `s : Set Ω` measurable,
```

Comment:
You don't need `s` to be measurable. Here is a proof (deducing it from the measurable case that you've already done).
```
lemma condKernel_apply_of_ne_zero' [MeasurableSingletonClass α] {x : α} (hx : ρ.fst {x} ≠ 0)
    {s : Set Ω} : ρ.condKernel x s = (ρ.fst {x})⁻¹ * ρ ({x} ×ˢ s) := by
  have A : MeasurableEmbedding (Prod.mk x : Ω → α × Ω) := by
    refine ⟨Prod.mk.inj_left x, measurable_prod_mk_left, ?_⟩ 
    intro s hs
    have : MeasurableSet ({x} ×ˢ s) := (MeasurableSet.singleton x).prod hs
    convert this using 1
    simp
  have : ρ.condKernel x s = ((ρ.fst {x})⁻¹ • ρ).comap (fun (y : Ω) ↦ (x, y)) s := by
    congr 2
    ext s hs
    simp [condKernel_apply_of_ne_zero ρ hx hs]    
    rw [MeasurableEmbedding.comap_apply A]
    rfl
  simp [this, MeasurableEmbedding.comap_apply A, hx]
```
(The fact that `Prod.mk` is a measurable embedding should probably be its own lemma, I was surprised not to find it).

Reply:
Thanks! I added MeasurableEmbedding lemmas about Prod.mk